windows-agent/internal/proservices/landscape/connection.go

```diff
@@ -50,6 +50,20 @@ func newConnectionSettings(c landscapeHostConf) connectionSettings {
 // handshake is complete.
 func newConnection(ctx context.Context, d serviceData) (conn *connection, err error) {
 	defer decorate.OnError(&err, "could not connect to Landscape server")
+	retryPolicy := `{
+	"methodConfig": [{
+		"name": [{"service": "landscapehostagentapi.LandscapeHostAgent"}],
+		"waitForReady": false,
```

Comment:
I suspect we want this to be true, but tests need to change for that not to be a problem.

Running with `waitForReady: false` fills the log with a bunch of attempts to connect to an unresponsive Landscape server, thus no exponential backoff exists.

Running with `waitForReady: true` allows the gRPC framework to handle the failed RPCs transparently. I think the logs below behave as we want:

```diff 
time=2024-06-12T17:18:20-03:00 level=debug msg=[[Ubuntu-24.04:729576]] Request /agentapi.WSLInstance/Connected done
time=2024-06-12T17:18:20-03:00 level=warning msg=Landscape: stopped listening for commands: could not receive commands: rpc error: code = Unavailable desc = error reading from server: EOF
+ time=2024-06-12T17:18:20-03:00 level=warning msg=Landscape: connection dropped unexpectedly                                                                                                                       
+ time=2024-06-12T17:18:20-03:00 level=info msg=Landscape: connecting                                                                                                                                              time=2024-06-12T17:24:00-03:00 level=info msg=New connection from client [[UbuntuDev:498555]]
time=2024-06-12T17:24:00-03:00 level=info msg=New connection from client [[UbuntuDev:930361]]
time=2024-06-12T17:24:00-03:00 level=info msg=New connection from client [[UbuntuDev:959997]]                                                                                                                    time=2024-06-12T17:24:00-03:00 level=debug msg=[[UbuntuDev:959997]] New request /agentapi.WSLInstance/ProAttachmentCommands                                                                                      time=2024-06-12T17:24:00-03:00 level=debug msg=[[UbuntuDev:498555]] New request /agentapi.WSLInstance/LandscapeConfigCommands
time=2024-06-12T17:24:00-03:00 level=debug msg=[[UbuntuDev:930361]] New request /agentapi.WSLInstance/Connected                                                                                                  time=2024-06-12T17:24:04-03:00 level=debug msg=[[UbuntuDev:498555]] Requesting with parameters: Data: 0xc0001e8080                                                                                               time=2024-06-12T17:24:04-03:00 level=debug msg=[[UbuntuDev:959997]] Requesting with parameters: Data: 0xc00039c740                                                                                               time=2024-06-12T17:24:04-03:00 level=debug msg=[[UbuntuDev:930361]] Requesting with parameters: WslName: UbuntuDev, Id: ubuntu, VersionId: 24.04, PrettyName: Ubuntu 24.04 LTS, ProAttached: true, Hostname: mib01
time=2024-06-12T17:24:04-03:00 level=debug msg=[[UbuntuDev:930361]] Database: cache miss, creating "UbuntuDev" and adding it to the database                                                                     time=2024-06-12T17:24:04-03:00 level=debug msg=Distro "UbuntuDev": starting task processing
time=2024-06-12T17:24:04-03:00 level=debug msg=[[UbuntuDev:930361]] connection to Linux-side WSL service established
time=2024-06-12T17:25:54-03:00 level=info msg=New connection from client [[Ubuntu-24.04:539591]]
time=2024-06-12T17:25:54-03:00 level=debug msg=[[Ubuntu-24.04:539591]] New request /agentapi.WSLInstance/LandscapeConfigCommands
time=2024-06-12T17:25:54-03:00 level=info msg=New connection from client [[Ubuntu-24.04:551751]]
time=2024-06-12T17:25:54-03:00 level=debug msg=[[Ubuntu-24.04:551751]] New request /agentapi.WSLInstance/ProAttachmentCommands
time=2024-06-12T17:25:54-03:00 level=info msg=New connection from client [[Ubuntu-24.04:969859]]
time=2024-06-12T17:25:54-03:00 level=debug msg=[[Ubuntu-24.04:969859]] New request /agentapi.WSLInstance/Connected                                                                                               time=2024-06-12T17:26:01-03:00 level=debug msg=[[Ubuntu-24.04:539591]] Requesting with parameters: Data: 0xc0001e8300
time=2024-06-12T17:26:01-03:00 level=debug msg=[[Ubuntu-24.04:969859]] Requesting with parameters: WslName: Ubuntu-24.04, Id: ubuntu, VersionId: 24.04, PrettyName: Ubuntu 24.04 LTS, ProAttached: true, Hostname: mib01                                                                                                                                                                                                          time=2024-06-12T17:26:01-03:00 level=debug msg=[[Ubuntu-24.04:969859]] Database: cache hit. Overwriting properties for "Ubuntu-24.04"                                                                            time=2024-06-12T17:26:01-03:00 level=debug msg=[[Ubuntu-24.04:551751]] Requesting with parameters: Data: 0xc000606030
time=2024-06-12T17:26:01-03:00 level=debug msg=Distro "Ubuntu-24.04": starting task "tasks.ProAttachment task with token: C1*************************Wa"
time=2024-06-12T17:26:01-03:00 level=debug msg=[[Ubuntu-24.04:969859]] connection to Linux-side WSL service established                                                                                          time=2024-06-12T17:26:02-03:00 level=debug msg=Distro "Ubuntu-24.04": distro is running.                                                                                                                         time=2024-06-12T17:26:02-03:00 level=debug msg=Distro "Ubuntu-24.04": ensuring active connection.                                                                                                                time=2024-06-12T17:26:02-03:00 level=debug msg=Distro "Ubuntu-24.04": connection is active.                                                                                                                      time=2024-06-12T17:26:04-03:00 level=debug msg=[[Ubuntu-24.04:551751]] Requesting with parameters: Data: 0xc000042710
time=2024-06-12T17:26:04-03:00 level=error msg=distro "Ubuntu-24.04": task "tasks.ProAttachment task with token: C1*************************Wa" failed: failed but will be retried: pro detach: command returned error: : Failed running command '/snap/bin/canonical-livepatch disable' [exit(1)]. Message: system_key.go:149: cannot determine seccomp compiler version in generateSystemKey: signal: bad system call
cannot join mount namespace of pid 1: Operation not permitted

time=2024-06-12T17:26:10-03:00 level=debug msg=[[Ubuntu-24.04:969859]] Requesting with parameters: WslName: Ubuntu-24.04, Id: ubuntu, VersionId: 24.04, PrettyName: Ubuntu 24.04 LTS, ProAttached: true, Hostname: mib01
+ time=2024-06-12T17:26:49-03:00 level=debug msg=Landscape: starting handshake
time=2024-06-12T17:26:49-03:00 level=debug msg=Landscape: sending info: {state:{NoUnkeyedLiterals:{} DoNotCompare:[] DoNotCopy:[] atomicMessageInfo:<nil>} sizeCache:0 unknownFields:[] Token:C1*************************Wa Uid:b067a1834ace42a9991c3cf405c49ce7 Hostname:mib01 Instances:[id:"Client24.04" name:"mib01" version_id:"24.04" id:"Ubuntu-24.04" name:"mib01" version_id:"24.04" instance_state:Running id:"UbuntuWork3" name:"mib01" version_id:"24.04" id:"UbuntuDev" name:"mib01" version_id:"24.04" instance_state:Running] AccountName:standalone RegistrationKey:<nil> DefaultInstanceId:0xc0000907b0}
time=2024-06-12T17:26:49-03:00 level=info msg=Landscape: handshake completed
time=2024-06-12T17:26:49-03:00 level=info msg=Landscape: connected 
```